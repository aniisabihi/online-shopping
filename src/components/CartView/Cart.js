// noinspection JSUnresolvedVariable

import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import {
  addToCart,
  clearCart,
  decreaseCart,
  getTotals,
  removeFromCart,
} from "../../store/reducers/slices/cartSlice";
import { ToastContainer } from "react-toastify";

export default function Cart() {
  const cart = useSelector((state) => state.cart);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getTotals());
  }, [cart, dispatch]);

  const handleAddToCart = (product) => {
    dispatch(addToCart(product));
  };
  const handleDecreaseCart = (product) => {
    dispatch(decreaseCart(product));
  };
  const handleRemoveFromCart = (product) => {
    dispatch(removeFromCart(product));
  };
  const handleClearCart = () => {
    dispatch(clearCart());
  };

  return (
    <div className="detail-item">
      <Link to="/" className="return waves-effect waves-light">
        <i id="return-button" className="material-icons">
          arrow_back
        </i>
      </Link>

      <div id="product" className="card">
        <div className="cart-container">
          {cart.cartItems.length === 0 ? (
            <div className="cart-empty">
              <h2>Your cart is empty</h2>
            </div>
          ) : (
            <div>
              <div className="titles">
                <h3 className="product-title">Product</h3>
                <h3 className="price">Price</h3>
                <h3 className="quantity">Quantity</h3>
                <h3 className="total">Total</h3>
              </div>
              <div className="cart-items">
                {cart.cartItems &&
                  cart.cartItems.map((cartItem) => (
                    <div className="cart-item" key={cartItem.id}>
                      <div className="cart-product">
                        <div>
                          <h5>{cartItem.name}</h5>
                          <p>Color: {cartItem.color.value}</p>
                          {cartItem.power.value > 0 && (
                            <p>Power: {cartItem.power.value}</p>
                          )}
                          {cartItem.storage.value > 0 && (
                            <p>Storage: {cartItem.storage.value}</p>
                          )}
                          <button
                            onClick={() => handleRemoveFromCart(cartItem)}
                          >
                            Remove
                          </button>
                        </div>
                      </div>
                      <div className="cart-product-price">{cartItem.price}</div>
                      <div className="cart-product-quantity">
                        <button onClick={() => handleDecreaseCart(cartItem)}>
                          -
                        </button>
                        <div className="count">{cartItem.cartQuantity}</div>
                        <button onClick={() => handleAddToCart(cartItem)}>
                          +
                        </button>
                      </div>
                      <div className="cart-product-total-price">
                        {cartItem.price * cartItem.cartQuantity} SEK
                      </div>
                    </div>
                  ))}
              </div>
              <div className="cart-summary">
                <button className="clear-btn" onClick={() => handleClearCart()}>
                  Clear Cart
                </button>
                <div className="cart-checkout">
                  <div className="subtotal">
                    <span>Total</span>
                    <span className="amount">{cart.cartTotalAmount} SEK</span>
                  </div>
                </div>
              </div>
              <ToastContainer />
            </div>
          )}
        </div>
      </div>
    </div>
  );
}
