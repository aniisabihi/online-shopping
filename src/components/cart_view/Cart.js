import React from "react";
import { Link } from "react-router-dom";
import ErrorBoundary from "../error_handling/ErrorBoundary";
import { Quantity } from "../../slices/Quantity";

export default function Cart() {
  // noinspection JSUnresolvedVariable

  return (
    <div className="detail-item">
      <Link to="/" className="return waves-effect waves-light">
        <i id="return-button" className="material-icons">
          arrow_back
        </i>
      </Link>

      <div id="product" className="card">
        <div className="card-content">Product</div>

        <div id="product-information" className="card-information">
          <h5>
            <b>Namn</b>
          </h5>

          <h6>Brand</h6>

          <div className="row">
            <div className="col left">left</div>
            <div className="col right">
              <ErrorBoundary>
                <Quantity />
              </ErrorBoundary>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
