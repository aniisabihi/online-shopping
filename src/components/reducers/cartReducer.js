import items from './../../products.json';

const initState = {
    items: [items],
    addedItems: [],
    total: 0
}

const cartReducer= (state = initState, action) => {

    return state;
}

export default cartReducer;
