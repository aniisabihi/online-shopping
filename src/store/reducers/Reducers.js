import { combineReducers } from "redux";
import cartReducer from "./slices/cartSlice";

const data = require("./../../data.json");

export const reducers = combineReducers({
  products: data,
  cart: cartReducer,
});

export default data;
