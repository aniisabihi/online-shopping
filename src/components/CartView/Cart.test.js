import React from "react";
import { shallow } from "enzyme";
import Cart from "./Cart";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { reducers } from "../../store/reducers/Reducers";

const CartComponent = <Cart />;
const store = configureStore({ reducer: reducers });

let cart = shallow(<Provider store={store}>{CartComponent}</Provider>);

/* RENDER COMPONENT */
describe("Render Cart", () => {
  it("should render the Cart Component accordingly", () => {
    expect(cart).toMatchSnapshot();
  });
});
