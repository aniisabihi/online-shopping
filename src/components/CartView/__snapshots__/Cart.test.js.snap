// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Cart should render the Cart Component accordingly 1`] = `ShallowWrapper {}`;
